### ISSUE TITLE
BatchNorm can not converge with scale=False

### ISSUE THREAD

**Issue by nttstar** [apache/incubator-mxnet#18475]
## Description
 BatchNorm operator with ``scale=False`` can not converge.

### Error Message
No error message, but loss value and training accuracy is abnormal comparing with ``scale=True`` BatchNorm.

## To Reproduce
We can try ``https://github.com/nttstar/arcface.np`` to train arcface. Add one BatchNorm op with ``scale=False`` after final embedding layer

## What have you tried to solve it?

1. Set ``Scale=True``, it can work but with slightly worse test accuracy.

## Environment

----------Python Info----------
Version      : 3.6.9
Compiler     : GCC 7.3.0
Build        : ('default', 'Jul 30 2019 19:07:31')
Arch         : ('64bit', '')
------------Pip Info-----------
Version      : 19.3.1
Directory    : /root/anaconda2/envs/py36/lib/python3.6/site-packages/pip
----------MXNet Info-----------
Version      : 2.0.0
Directory    : /root/anaconda2/envs/py36/lib/python3.6/site-packages/mxnet
Num GPUs     : 8
Hashtag not found. Not installed from pre-built package.
----------System Info----------
Platform     : Linux-3.10.0-327.el7.x86_64-x86_64-with-centos-7.5.1804-Core
system       : Linux
node         : gpu06
release      : 3.10.0-327.el7.x86_64
version      : #1 SMP Thu Nov 19 22:10:57 UTC 2015

**Comment by wkcn** [comment]
Hi @nttstar , there was a bug in BatchNorm in the previous version of MXNet. The bug was reported in #18373 , and it was fixed in PR #18377 . Could you please try the latest version of MXNet?

**Comment by sxjscience** [comment]
@wkcn I communicated with @nttstar offline and https://github.com/apache/incubator-mxnet/issues/18373 should not be the casue. Would you help try with `scale=False` and see if there is anything wrong?

**Comment by wkcn** [comment]
@sxjscience I'm sorry that I don't have any machine with GPU to check it recently.

I read the code of batch norm and its unittest.
There is a gradient check when `fix_gamma=True`:

https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py#L1777, 

but no output check when `fix_gamma=True`: 

https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py#L1882

**Comment by wkcn** [comment]
I try to test the batch norm with `fix_gamma=True` on GPU, the test fails on `assert_almost_equal(bn_beta.grad.asnumpy(), db.asnumpy(), atol=atol, rtol=rtol)` in [test_operator.py#L1950](https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py#L1950).

```
(<function BatchNorm at 0x7f54854753b0>, (24, 2), 0, False)
(<function BatchNorm at 0x7f54854753b0>, (24, 2), 0, False)
(<function BatchNorm at 0x7f54854753b0>, (24, 2), 0, True)
(<function BatchNorm at 0x7f54854753b0>, (24, 2), 0, True)
(<function BatchNorm at 0x7f54854753b0>, (24, 2), 1, False)

*** Maximum errors for vector of size 2:  rtol=0.01, atol=0.01

  1: Error 90.669586  Location of error: (0,), a=28.82106400, b=14.64017868
  2: Error 64.738541  Location of error: (1,), a=22.83149147, b=13.46625137
```

The result on CPU is correct.

**Comment by wkcn** [comment]
I'm fixing the bug and I will submit a PR later.

**Comment by wkcn** [comment]
Hi @nttstar , the bug of BatchNorm has been solved in #18492 .

**Comment by nttstar** [comment]
@wkcn Thanks! I will check it in the next pip package release.
